charts/consul/templates/connect-inject-clusterrole.yaml

```diff
@@ -24,6 +24,13 @@ rules:
   - "get"
   - "list"
   - "watch"
+- apiGroups: [ "" ]
+  resources: ["secrets"]
+  verbs:
+    - "get"
+    - "list"
+    - "watch"
+    - "create"
```

Comment:
Ahh yes! The unit tests didn't catch that but I think once I manually test the secret deletion case it will! 

For the ownerRef garbage collection, we don't need delete access, but for the case where the secret name was changed we will!